### PR TITLE
Fix issue 2256 - doc grammar and typos in protocol-spec.md

### LINKED ISSUE #2256: Some editorials for protocol spec
## Description

Suggested editorials for protocol spec. Some are typos. Encountered while reading the security sections for my own understanding.
## Describe How to Reproduce

View the protocol spec section listed below, at: 
https://github.com/hyperledger/fabric/blob/master/docs/protocol-spec.md

4.5 Online wallet service - (question: we have an online wallet in the "current release"? 4.7 covers "restrictions in this release" so that is the opportunity to clarify current vs. next support for an online wallet.)
4.7 bullet 2: chain-code >> chaincode
4.7 bullet 2: any-one >> anyone
4.7 bullet 5: if the application ignores to bind the transaction binding offered by our fabric, >> if the application ignores the transaction binding offered by our fabric, 
4.7.1: role of an online wallet. >> (question: we have an "online wallet" in this "release"?) 
4.7.1: non-validating peer who plays >> non-validating peer that plays (i.e. avoid personification of a node / non-human entity)
4.7.1: the end-user leverages his registration >> the end-user leverages their registration (i.e. use the "singular their" to avoid gendered attributions.) 
4.7.2: (Might help to clarify what the "content of a transaction" is exactly, and what "content" is confidential, or point to that).
4.7.2 auditors that are not member of the system >> auditors who are not members of the blockchain network 
4.7.2 the Blockchain data, >> the blockchain data (and >> blockchain ledger? ... if the only auditable "data" is the ledger.) 
4.7.2 aside the transactions themselves, also includes the state updates of individual contracts or chaincodes. >> aside from the transactions themselves ... (question: are we using "smart contracts" as synonymous with "chaincodes" ? ... would using "smart contracts" alone be more precise when we mean that; because the ledger contains other "chaincode" (transactions) that are not smart contracts.)
4.7.2 Membership service generates >> Membership Services generates
4.7.2 At enrollment phase, user >> At the enrollment phase, each user 
4.7.2 while each validator vj obtain its enrollment certificate denoted >> while each validator vj obtains its enrollment certificate, denoted

### PR DESCRIPTION
**Pull request by christo4ferris**
## Description

Fix issue 2256 - doc grammar and typos in protocol-spec.md
## Motivation and Context

Fixes #2256 
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [x] I have either added documentation to cover my changes or this change requires no new documentation.
- [x] I have either added unit tests to cover my changes or this change requires no new tests.
- [] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Christopher Ferris chrisfer@us.ibm.com

### CODE REVIEW

File: docs/protocol-spec.md [hunk 1]
```diff
@@ -2126,7 +2126,7 @@ the code-metadata fields near it), and provide those to containers for deploymen
 ### 4.5 Online wallet service
 
 
-This section describes the security design of a wallet service, which in this case is a node where end-users can register, move their key material to, and perform transactions through.
+This section describes the security design of a wallet service, which in this case is a node with which end-users can register, store their key material, and through which they can perform transactions.
```
Comment (JonathanLevi) [inline docs/protocol-spec.md]: their key **material and** through which they can perform transactions.

no `,`

File: docs/protocol-spec.md [hunk 2]
```diff
@@ -2195,9 +2195,9 @@ as depicted in Sections 4.7.1 and 4.7.2.
  - Client side enrollment and transaction creation is performed entirely by a
    non-validating peer that is trusted not to impersonate the user.
    See, Section 4.7.1 for more information.
- - A minimal set of confidentiality properties where a chain-code is accessible
+ - A minimal set of confidentiality properties where a chaincode is accessible
    by any entity that is member of the system, i.e., validators and users who
-   have registered to our membership services, and not accessible by any-one else.
+   have registered to our membership services, and not accessible by anyone else.
```
Comment (JonathanLevi) [inline docs/protocol-spec.md]: have registered **through Hyperledger Fabric's Membership Services** and not ...
(through or using)

Comment (JonathanLevi) [inline docs/protocol-spec.md]: have registered **through Hyperledger Fabric's Membership Services** and **is** not accessible by anyone else

File: docs/protocol-spec.md [hunk 3]
```diff
@@ -2207,14 +2207,14 @@ as depicted in Sections 4.7.1 and 4.7.2.
...
-   processing  may be at risk.
+   for security to be guaranteed. This means, that if the application fails to
+   *bind* the transaction binding offered by the fabric, secure transaction
+   processing may be at risk.
 
 #### 4.7.1 Simplified client
 
-Client side enrollment and transaction creation is performed entirely by a non-validating peer who plays the role of an online wallet.
-In particular, the end-user leverages his registration credentials <username, password> to open an account to a non-validating peer
+Client side enrollment and transaction creation is performed entirely by a non-validating peer that plays the role of an online wallet.
```
Comment (JonathanLevi) [inline docs/protocol-spec.md]: **Client-side** enrollment and transaction creation **are** performed entirely by a non-validating peer **which** plays the role of an online wallet.

Comment (christo4ferris) [inline docs/protocol-spec.md]: actually, I disagree on the use of which where that is appropriate. Which is used when there is a choice. all other changes made. thanks for the ocd review;-)

**Comment by christo4ferris** [comment]
@srderson think that this can be merged, now.

**Comment by srderson** [comment]
LGTM, thanks

**Pull request merged**